### PR TITLE
Install tags-system in the application

### LINKED ISSUE #343: Install pg
install pg as dependency, so that we can share a pool across all our plugins.

### PR DESCRIPTION
**Pull request by mk4111**
Closes #343, #344, #345

adds local config, registers plugin, changes tags format and IDs
affects multiple files

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -21,7 +25,15 @@ function initServer (config, callback) {
...
-    auth
+    auth,
+    {
+      register: tags,
+      options: {
+        tags: tagsData,
+        categories: categoriesData,
+        pool: pool
+      }
+    }
```
Comment (JMurphyWeb) [inline server/server.js]: Maybe not necessary now, but after 2 or more custom plugins, this method of registering is going to get ugly. We can export it into another module that only handles registering this plugin?

Comment (SimonLab) [inline server/server.js]: At the moment it's not a problem, we can create an issue for that if we want to enhance the registration plugin process

File: server/start.js [hunk 2]
```diff
@@ -4,11 +4,6 @@ var Hoek = require('hoek');
 var initServer = require('./server.js');
 var config = require('./config.js');
 
-// set up db
-require('../test/helpers/set-up.js').initialiseDB(function () {
```
Comment (JMurphyWeb) [inline server/start.js]: as discussed, some data is needed

Comment (SimonLab) [inline server/start.js]: Resetting automatically the data each the server start will create some issue when the application is online as we won't have any persistent data anymore. If we need to clean the data we should do this automatically see #347

**Pull request merged**